pkg/apis/core/validation/validation.go

```diff
@@ -1243,13 +1243,44 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 		return allErrs
 	}
 
-	privileged := container.SecurityContext != nil && container.SecurityContext.Privileged != nil && *container.SecurityContext.Privileged
-	if *mountPropagation == core.MountPropagationBidirectional && !privileged {
-		allErrs = append(allErrs, field.Forbidden(fldPath, "Bidirectional mount propagation is available only to privileged containers"))
+	if *mountPropagation == core.MountPropagationBidirectional && !hasBidirectionalMountPropagationPermissions(container.SecurityContext, opts) {
+		errMessage := "Bidirectional mount propagation is available only to privileged containers"
+		if opts.AllowUnprivilegedBidirectionalMountPropagation {
+			errMessage += " or containers with the 'SYS_ADMIN' capability"
```

Comment:
@enj brought up an interesting point that this could potentially be a backwards incompatible change for security policy enforcement. For example, someone could have intended to restrict mount propagation behavior via a privileged pod policy, and with this change, that policy would no longer be effective and they would have to also add a policy looking at SYS_ADMIN capabilities.

We should probably discuss this further with sig-auth to more deeply understand how powerful SYS_ADMIN without privileged is today and would we be giving substantial host/root capabilities to pods with bidirectional mount propagation.

Reply:
How can I/we start the discussion with sig-auth on this topic?

I can prepare something to outline the CSI driver use case, but I'm unsure what else may be needed.

Reply:
The best is to add an item to their regular meeting agenda + discuss it there (+ link this PR)
 https://github.com/kubernetes/community/blob/master/sig-auth/README.md

Reply:
Discussed this topic with @enj and others in the [sig-auth meeting on 2nd august](https://docs.google.com/document/d/1woLGRoONE3EBVx-wTb4pvp4CI7tmLZ6lS26VTbosLKM/edit#heading=h.jt7x40x5om30):
* `SYS_ADMIN` capability is still very powerful without privileged.
* PSA baseline should be sufficient to block unwanted configuration, as `SYS_ADMIN` capability is only allowed in privileged PSS.
* Should be fine to allow `SYS_ADMIN` capability for unprivileged containers because of these restrictions.

Open question: How can additional requirements for `mount` operation be validated, e.g. app armor?
- When app armor is enabled on the node,`unconfined` or custom profile must be configured for the container via annotation.
- This annotation is not necessary if the app armor feature is not enabled on the node.